src/systems/settling.cairo

```diff
@@ -0,0 +1,289 @@
+#[system]
+mod Settle {
+    use traits::Into;
+    use box::BoxTrait;
+
+    use eternum::utils::unpack::unpack_resource_ids;
+    use eternum::constants::WORLD_CONFIG_ID;
+    use eternum::interfaces::{IERC721Dispatcher, IERC721DispatcherTrait};
+    use eternum::erc721::erc721::{RealmData, Position};
+    use eternum::components::owner::Owner;
+    use eternum::components::realm::Realm;
+    use eternum::components::resources::Resource;
+    use eternum::components::age::Age;
+    use eternum::components::config::WorldConfig;
+
+    fn execute(realm_id: felt252) { // get the ERC721 contract
+        // get the owner
+        let config = commands::<WorldConfig>::entity(WORLD_CONFIG_ID.into());
+        let token = config.realm_l2_contract;
+        let owner = commands::<Owner>::entity((token, realm_id).into());
+        // TODO: wait for set_account_contract_address to be available
+        // let caller = starknet::get_tx_info().unbox().account_contract_address;
+        // TODO: withdraw gas error with assert 
+        // assert(owner.address == caller, 'Only owner can settle');
+        // get the metadata
+        let erc721 = IERC721Dispatcher { contract_address: token };
+        let realm_data: RealmData = erc721.fetch_realm_data(realm_id);
+        let position: Position = erc721.realm_position(realm_id);
+        // create Realm Metadata
+        let realm_query: Query = realm_id.into();
+        commands::set_entity(
+            realm_query,
+            (
+                Position {
+                    x: position.x, y: position.y, 
+                    }, Realm {
+                    realm_id,
+                    owner: owner.address,
+                    resource_ids_packed: realm_data.resource_ids_packed,
+                    resource_ids_count: realm_data.resource_ids_count,
+                    cities: realm_data.cities,
+                    harbors: realm_data.harbors,
+                    rivers: realm_data.rivers,
+                    regions: realm_data.regions,
+                    wonder: realm_data.wonder,
+                    order: realm_data.order,
+                    }, Owner {
+                    address: owner.address, 
```

Comment:
Follow-up on the discussion above, the address here should not be set to `owner.address` but rather the tx_info account address.

Reply:
not sure why that wouldn't work though? 
the idea is to do this:
```
let owner = commands::<Owner>::entity((token, realm_id.into()).into());
let caller = starknet::get_tx_info().unbox().account_contract_address;
assert(owner.address == caller, 'Only owner can settle');
```
So owner.address and tx_info().account_contract_address should be the same thing here. I had to comment it out though because it will only be available in next release

Reply:
But there's no Owner entity yet, is there? The entity has not been set, so `commands::<Owner>::entity((token, realm_id.into()).into())` returns an empty/zero owner if I'm not mistaken.

Reply:
Owner is set during mint 
```
let token: felt252 = config.realm_l2_contract.into();
let owner = commands::<Owner>::entity((token, realm_id.into()).into());
```
this retrieves entity defined by ERC721

Reply:
Resolving to unlock merging